### PR TITLE
[docs] Use `Base UI` `Portal` for the quick filter recipe

### PR DESCRIPTION
**Pull request by DanailH**
- [ ] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

Based on https://github.com/mui/mui-x/pull/10121#discussion_r1310683314

### CODE REVIEW

File: docs/data/data-grid/filtering-recipes/QuickFilterOutsideOfGrid.js [hunk 1]
```diff
@@ -8,10 +8,9 @@ import { useDemoData } from '@mui/x-data-grid-generator';
 function MyCustomToolbar(props) {
   return (
     <React.Fragment>
-      {ReactDOM.createPortal(
-        <GridToolbarQuickFilter />,
-        document.getElementById('filter-panel'),
-      )}
+      <Portal container={() => document.getElementById('filter-panel')}>
```
Comment (MBilalShafi) [inline docs/data/data-grid/filtering-recipes/QuickFilterOutsideOfGrid.js:11]: Could we directly use the selector here?
```suggestion
      <Portal container={document.getElementById('filter-panel')}>
```

As per docs:
image

File: docs/data/data-grid/filtering-recipes/QuickFilterOutsideOfGrid.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import * as React from 'react';
-import * as ReactDOM from 'react-dom';
+import { Portal } from '@mui/base/Portal';
```
Comment (MBilalShafi) [inline docs/data/data-grid/filtering-recipes/QuickFilterOutsideOfGrid.js:2]: Would `export default` usage be more appropriate as the path being used is an absolute one?
```suggestion
import Portal from '@mui/base/Portal';
```

**Review by MBilalShafi — APPROVED** [review]

File: docs/data/data-grid/filtering-recipes/QuickFilterOutsideOfGrid.js (hunk 2 above)
Comment (oliviertassinari) [reply docs/data/data-grid/filtering-recipes/QuickFilterOutsideOfGrid.js:2]: This recently changed https://github.com/mui/material-ui/releases/tag/v5.14.4. As I understand it, the idea is to have this for all @mui packages, as each gets a new major. cc @michaldudak

File: docs/data/data-grid/filtering-recipes/QuickFilterOutsideOfGrid.js (hunk 1 above)
Comment (oliviertassinari) [reply docs/data/data-grid/filtering-recipes/QuickFilterOutsideOfGrid.js:11]: It would break during server side rendering, I guess.

**Pull request merged**